### PR TITLE
fix fatal error when translation key is numeric

### LINKED ISSUE #34603: fatal error when translation key is numeric
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When using legacy translation system and the key is numeric e.g. 

1 => "1" 
2 => "2" 

The whole translation system breaks because md5 cannot handle numbers. 

283630911-728baad8-a23b-4a6d-a834-8387a91bf652

### Expected behavior

Translation system works

### Steps to reproduce

Create a translation file with numeric keys as index

### PrestaShop version(s) where the bug happened

1.7.8.9

### PHP version(s) where the bug happened

7.4+

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

IOWEB TECHNOLOGIES

### PR DESCRIPTION
**Pull request by ioweb-gr**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This is affecting every prestashop version after 1.7.8.9 (tested after this version). The md5 function will throw an error if you feed it with a numeric value. This will cause the system to be unable to load the backoffice translations page. e.g. the url admin/index.php/api/translations/tree/el/themes/.....  will throw exception
| Type?             | bug fix 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     |  no
| How to test?      | Add a translation file with the legacy system using a numeric key. 
| UI Tests          | https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/6901292359
| Fixed issue or discussion?     | #34603 
| Related PRs       | None
| Sponsor company   | IOWEB TECHNOLOGIES

Exception thrown

```
Type error: md5() expects parameter 1 to be string, int given

[Symfony\Component\Debug\Exception\FatalThrowableError 0]

```

image

### CODE REVIEW

**Comment by ioweb-gr** [comment]
I'm not sure how to run the UI tests, there's no action tab when I fork the repo described in the docs.

**Comment by nicosomb** [comment]
@ioweb-gr the Actions tab is here: https://github.com/ioweb-gr/ga.tests.ui.pr/actions


I launched the UI tests for you https://github.com/nicosomb/ga.tests.ui.pr/actions/runs/6901292359

**Comment by nicosomb** [comment]
@ioweb-gr is there an opened issue for this fix?

**Comment by ioweb-gr** [comment]
@nicosomb Yeah this is the issue https://github.com/PrestaShop/PrestaShop/issues/34603

I actually couldn't see the actions tab. When I clicked on your link I had to enable workflows to see it

File: src/Core/Translation/Storage/Provider/ModuleCatalogueLayersProvider.php [hunk 1]
```diff
@@ -303,7 +303,7 @@ private function buildTranslationCatalogueFromLegacyFiles(string $locale): Messa
 
         foreach ($catalogueFromPhpAndSmartyFiles->all() as $currentDomain => $items) {
             foreach (array_keys($items) as $translationKey) {
-                $legacyKey = md5($translationKey);
+                $legacyKey = md5((string)$translationKey);
```
Comment (matthieu-rolland) [inline src/Core/Translation/Storage/Provider/ModuleCatalogueLayersProvider.php]:
```suggestion
                $legacyKey = md5((string) $translationKey);
```
This should fix the csfixer issue

**Review by matthieu-rolland — COMMENTED** [review]
LGTM, just one code suggestion to fix the csfixer issue

**Comment by ioweb-gr** [comment]
I'll adjust it later

File: src/Core/Translation/Storage/Provider/ModuleCatalogueLayersProvider.php (hunk 1 above)
Comment (ioweb-gr) [reply src/Core/Translation/Storage/Provider/ModuleCatalogueLayersProvider.php]: Done @matthieu-rolland

**Review by mflasquin — APPROVED** [review]

**Review by boherm — APPROVED** [review]
Thanks @ioweb-gr!

**Review by matthieu-rolland — APPROVED** [review]
thanks

**Review by ShaiMagal — APPROVED** [review]

**Comment by tleon** [comment]
Hello I wanted to test this PR. Could you provide me with a little more detailed step to test this ? 

Thanks in advance. 

BR

**Comment by florine2623** [comment]
Friendly reminder @ioweb-gr 

Could you share the steps to reproduce please ? 
Thanks!

**Comment by ioweb-gr** [comment]
@florine2623 I'm not sure exactly with which steps to replicate. I encountered it on my system from some module's translation file and it gave me the error. By debugging line by line I found the root cause which didn't handle it properly and patched it. 

There's nothing more than that I'm afraid.

**Comment by florine2623** [comment]
Maybe you could send a screen record of what you did until you encountered the exception ?
That would be very helpful !

**Comment by ioweb-gr** [comment]
Hi, unfortunately our client has decided to quit prestashop in the upcoming period therefore we no longer hold any interest in the platform. Feel free to close or merge the PR or continue on the issues we opened up. 

There are too many issues to count to be productive with Prestashop I'm afraid.

**Comment by florine2623** [comment]
Hello @ioweb-gr ,

Sorry to hear about this.
I'll close your PR, if you have more info about it and want your PR to be tested, do not hesitate to recreate another one.

Thanks!

**Comment by kpodemski** [comment]
For what is worth, [this change is already in place for PrestaShop 9](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Translation/Storage/Provider/ModuleCatalogueLayersProvider.php#L306).

**Pull request closed without merging**